hydroserving/client/contract.py

```diff
@@ -0,0 +1,374 @@
+from typing import Dict, Tuple
+import hydro_serving_grpc as hs_grpc
+import pandas as pd
+import yaml
+
+from google.protobuf.json_format import MessageToDict
+
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except ImportError:
+    from yaml import Loader, Dumper
+
+from .type_conversions import HS_TO_NP_DTYPE, NP_TO_HS_DTYPE, NP_DTYPE_TO_ARG_NAME, STR_TO_HS_DTYPE, HS_DTYPE_TO_STR
+import numpy as np
+
+
+class ContractViolationException(Exception):
+    """
+    Exception raised after failed compliance check
+    """
+    pass
+
+
+class HSContract:
+    """
+    HydroServingContract contains useful info about input/output
+    tensors and verifies passed dictionary for compliance with contract
+    """
+
+    def __str__(self) -> str:
+        return f"Input tensors = {self.input_names}; Output tensors = {self.output_names} "
+
+    @classmethod
+    def from_df(cls, input_df: pd.DataFrame):
+        """
+        Suggest contract definition for model contract from dataframe
+        :param input_df:
+        :return:
+        """
+        inputs = []
+        for name, dtype in zip(input_df.columns, input_df.dtypes):
+            inputs.append({"name": name,
+                           "dtype": dtype,
+                           "shape": (-1, 1)})
+
+        outputs = []
+
+        contract_dict = {"inputs": inputs,
+                         "outputs": outputs}
+        return cls(contract_dict)
+
+    @classmethod
+    def from_proto(cls, model_version_proto):
```

Comment:
what's model version proto?
why not contract proto?
why there is no `to_proto` conversion?